Kokkos_ENABLE_BENCHMARKS=ON configure fails on WSL due to HAVE_GPU_POSIX_REGEX failing

**Describe the bug**

When building in WSL (Ubuntu 22.04.3) with just `Kokkos_ENABLE_BENCHMARKS=ON` set, CMake fails at the `Performing Test HAVE_GNU_POSIX_REGEX` step.  From the CMake output below, it appears that Kokkos requires `HAVE_GPU_POSIX_REGEX` even though there is a `HAVE_STD_REGEX` option.

```
[benchmark] /home/mhoemmen/bld/kokkos/kokkos-2/_deps/googlebenchmark-src/cmake/gnu_posix_regex.cpp:1:10: fatal error: gnuregex.h: No such file or directory
[benchmark]     1 | #include <gnuregex.h>
```

`Kokkos_ENABLE_TESTS=ON Kokkos_ENABLE_EXAMPLES=ON` configures, builds, and passes tests just fine on this system.

**Please include the following for a minimal reproducer**

1. Compilers (with versions): GCC 11.4
2. Kokkos release or commit used (i.e., the sha1 number): e0dc0128e04f18c2bbbaefceef3616e7ddcfa3c4 (tag: 4.2.01)
3. Platform, architecture and backend: Ubuntu 22.04.3 LTS, WSL
4. CMake configure command: `cmake $HOME/src/kokkos/kokkos -DKokkos_ENABLE_BENCHMARKS=ON`
5. Output from CMake configure command

```
-- Setting default Kokkos CXX standard to 17
-- The CXX compiler identification is GNU 11.4.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Setting build type to 'RelWithDebInfo' as none was specified.
-- Kokkos version: 4.2.1
-- The project name is: Kokkos
-- Using internal gtest for testing
-- Configured git information in /home/mhoemmen/bld/kokkos/kokkos-2/generated/Kokkos_Version_Info.cpp
-- SERIAL backend is being turned on to ensure there is at least one Host space. To change this, you must enable another host execution space and configure with -DKokkos_ENABLE_SERIAL=OFF or change CMakeCache.txt
-- Using -std=gnu++17 for C++17 extensions as feature
-- Built-in Execution Spaces:
--     Device Parallel: NoTypeDefined
--     Host Parallel: NoTypeDefined
--       Host Serial: SERIAL
--
-- Architectures:
...
-- [benchmark] Performing Test HAVE_STD_REGEX
-- [benchmark] Performing Test HAVE_STD_REGEX
-- [benchmark] Performing Test HAVE_STD_REGEX -- success
-- [benchmark] Performing Test HAVE_GNU_POSIX_REGEX
-- [benchmark] Performing Test HAVE_GNU_POSIX_REGEX
-- [benchmark] Performing Test HAVE_GNU_POSIX_REGEX -- failed to compile: Change Dir: /home/mhoemmen/bld/kokkos/kokkos-2/CMakeFiles/CMakeTmp
[benchmark]
[benchmark] Run Build Command(s):/usr/bin/gmake -f Makefile cmTC_f8c5e/fast && /usr/bin/gmake  -f CMakeFiles/cmTC_f8c5e.dir/build.make CMakeFiles/cmTC_f8c5e.dir/build
[benchmark] gmake[1]: Entering directory '/home/mhoemmen/bld/kokkos/kokkos-2/CMakeFiles/CMakeTmp'
[benchmark] Building CXX object CMakeFiles/cmTC_f8c5e.dir/gnu_posix_regex.cpp.o
[benchmark] /usr/bin/c++   -std=c++11  -Wall  -Wextra  -Wshadow  -Wfloat-equal  -Wsuggest-override  -pedantic  -pedantic-errors  -fstrict-aliasing  -Wno-deprecated-declarations  -Wstrict-aliasing  -o CMakeFiles/cmTC_f8c5e.dir/gnu_posix_regex.cpp.o -c /home/mhoemmen/bld/kokkos/kokkos-2/_deps/googlebenchmark-src/cmake/gnu_posix_regex.cpp
[benchmark] /home/mhoemmen/bld/kokkos/kokkos-2/_deps/googlebenchmark-src/cmake/gnu_posix_regex.cpp:1:10: fatal error: gnuregex.h: No such file or directory
[benchmark]     1 | #include <gnuregex.h>
[benchmark]       |          ^~~~~~~~~~~~
[benchmark] compilation terminated.
[benchmark] gmake[1]: *** [CMakeFiles/cmTC_f8c5e.dir/build.make:78: CMakeFiles/cmTC_f8c5e.dir/gnu_posix_regex.cpp.o] Error 1
[benchmark] gmake[1]: Leaving directory '/home/mhoemmen/bld/kokkos/kokkos-2/CMakeFiles/CMakeTmp'
[benchmark] gmake: *** [Makefile:127: cmTC_f8c5e/fast] Error 2
[benchmark]
[benchmark]
```

6. Minimum, complete code needed to reproduce the bug: Just building Kokkos benchmarks.
7. Command line needed to reproduce the bug: See above
8. `KokkosCore_config.h` header file (generated during the build): CMake didn't finish
9. Please provide any additional relevant error logs: N/A

--- a/core/perf_test/CMakeLists.txt
+++ b/core/perf_test/CMakeLists.txt
@@ -50,8 +50,8 @@ ELSE()
   FetchContent_Declare(
     googlebenchmark
     DOWNLOAD_EXTRACT_TIMESTAMP FALSE
-    URL https://github.com/google/benchmark/archive/refs/tags/v1.6.2.tar.gz
-    URL_HASH MD5=14d14849e075af116143a161bc3b927b
+    URL https://github.com/google/benchmark/archive/refs/tags/v1.7.1.tar.gz
+    URL_HASH MD5=0459a6c530df9851bee6504c3e37c2e7
   )
   FetchContent_MakeAvailable(googlebenchmark)
   list(POP_BACK CMAKE_MESSAGE_INDENT)

